### PR TITLE
As/csw proof param constant

### PR DESCRIPTION
**Pull request by alsala**
related to zend_oo PR #179

### CODE REVIEW

File: examples/sampleCalls.cpp [hunk 1]
```diff
@@ -1010,10 +1017,10 @@ TEST_SUITE("Single Proof Verifier") {
...
 
         // Test all cases
-        create_verify_csw_proof(true, true, proof_path, pk_path, vk_path);
-        create_verify_csw_proof(true, false, proof_path, pk_path, vk_path);
-        create_verify_csw_proof(false, true, proof_path, pk_path, vk_path);
-        create_verify_csw_proof(false, false, proof_path, pk_path, vk_path);
+        create_verify_csw_proof(true, true, proof_path, pk_path, vk_path, true);
+        create_verify_csw_proof(true, false, proof_path, pk_path, vk_path, true);
+        create_verify_csw_proof(false, true, proof_path, pk_path, vk_path, true);
+        create_verify_csw_proof(false, false, proof_path, pk_path, vk_path, true);
```
Comment (DanieleDiBenedetto) [inline examples/sampleCalls.cpp:1023]: Can we add another test case "Proof Verifier: CSWNoConst - Coboundary Marlin" and pass the last flag to false (like I did for Cert) ?

File: examples/sampleCalls.cpp [hunk 2]
```diff
@@ -1044,10 +1051,10 @@ TEST_SUITE("Single Proof Verifier") {
...
```
Comment (DanieleDiBenedetto) [inline examples/sampleCalls.cpp:1057]: Can we add another test case "Proof Verifier: CSWNoConst -Darlin" and pass the last flag to false (like I did for Cert) ?

**Review by DanieleDiBenedetto — CHANGES_REQUESTED** [review]

**Review by i-Alex — CHANGES_REQUESTED** [review]

File: src/mc_test_circuits/csw.rs [hunk 3]
```diff
@@ -38,7 +63,9 @@ fn enforce_csw_inputs_gadget<CS: ConstraintSystem<FieldElement>>(
         || aggregated_input.ok_or(SynthesisError::AssignmentMissing)
     )?;
 
-    for i in 0..(num_constraints - 1)/4{
+    let remaining_constraints = if constant_present { (num_constraints - 1)/8 } else { (num_constraints - 1)/4 };
```
[the remark this answers is not in the document]
Comment (DanieleDiBenedetto) [reply src/mc_test_circuits/csw.rs:66]: yes the logic it's the same

File: src/mc_test_circuits/csw.rs [hunk 4]
```diff
@@ -119,20 +152,28 @@ impl ConstraintSynthesizer<FieldElement> for CSWTestCircuitWithAccumulators {
...
+) -> Result<(ZendooProverKey, ZendooVerifierKey), ProvingSystemError>
 {
     let ck_g1 = get_g1_committer_key()?;
     match ps {
         ProvingSystem::Undefined => Err(ProvingSystemError::UndefinedProvingSystem),
         ProvingSystem::Darlin => {
             let ck_g2 = get_g2_committer_key()?;
             let circ = CSWTestCircuitWithAccumulators {
+                constant_present: with_constant,
+                constant: None,
```
[the remark this answers is not in the document]
Comment (DanieleDiBenedetto) [reply src/mc_test_circuits/csw.rs:176]: When creating the keys, constant doesn't have a value so it's ok that it's None.

**Review by DanieleDiBenedetto — APPROVED** [review]

**Review by i-Alex — APPROVED** [review]

**Review by albertog78 — APPROVED** [review]

**Pull request merged**